core/dbt/parser/manifest.py

```diff
@@ -333,15 +332,214 @@ def get_full_manifest(
+        # the unique_id instead of the name. Sources are loaded from yaml files, so
+        # aren't in place yet
+        self.manifest.rebuild_ref_lookup()
+        self.manifest.rebuild_doc_lookup()
+        self.manifest.rebuild_disabled_lookup()
+
+        # Load yaml files
+        parser_types = [SchemaParser]  # type: ignore
+        for project in self.all_projects.values():
+            if project.project_name not in project_parser_files:
+                continue
+            self.parse_project(project, project_parser_files[project.project_name], parser_types)
+
+        self.cleanup_disabled()
+
+        self._perf_info.parse_project_elapsed = time.perf_counter() - start_parse_projects
+
+        # patch_sources converts the UnparsedSourceDefinitions in the
+        # Manifest.sources to SourceDefinition via 'patch_source'
+        # in SourcePatcher
+        start_patch = time.perf_counter()
+        patcher = SourcePatcher(self.root_project, self.manifest)
+        patcher.construct_sources()
+        self.manifest.sources = patcher.sources
+        self._perf_info.patch_sources_elapsed = time.perf_counter() - start_patch
+
+        # We need to rebuild disabled in order to include disabled sources
+        self.manifest.rebuild_disabled_lookup()
+
+        # copy the selectors from the root_project to the manifest
+        self.manifest.selectors = self.root_project.manifest_selectors
+
+        # inject any available external nodes
+        self.manifest.build_parent_and_child_maps()
+        external_nodes_modified = self.inject_external_nodes()
+        if external_nodes_modified:
+            self.manifest.rebuild_ref_lookup()
+
+        # update the refs, sources, docs and metrics depends_on.nodes
+        # These check the created_at time on the nodes to
+        # determine whether they need processing.
+        start_process = time.perf_counter()
+        self.process_sources(self.root_project.project_name)
+        self.process_refs(self.root_project.project_name, self.root_project.dependencies)
+        self.process_docs(self.root_project)
+        self.process_metrics(self.root_project)
+        self.process_saved_queries(self.root_project)
+        self.check_valid_group_config()
+        self.check_valid_access_property()
+
+        semantic_manifest = SemanticManifest(self.manifest)
+        if not semantic_manifest.validate():
+            raise dbt.exceptions.ParsingError("Semantic Manifest validation failed.")
+
+        # update tracking data
+        self._perf_info.process_manifest_elapsed = time.perf_counter() - start_process
+        self._perf_info.static_analysis_parsed_path_count = (
+            self.manifest._parsing_info.static_analysis_parsed_path_count
+        )
+        self._perf_info.static_analysis_path_count = (
+            self.manifest._parsing_info.static_analysis_path_count
+        )
+
     # This is where the main action happens
     def load(self) -> Manifest:
```

Comment:
Could you move this method up above "load_macros_and_tests" (or whatever the name will be)?  It's better to be able to read down the file in the approximate order of execution instead of having to jump back and forth.